nservicebus/messaging/transactions.md

```diff
@@ -7,94 +7,108 @@ tags:
 ### Transport transaction
 
-In this mode, if supported by the transport, the receive operation is wrapped in a native transaction. The same transaction is used when processing of the message results in sending out other messages. This mode guarantees that the message is not permanently deleted from the incoming queue until at least one processing attempt (including storing user data and sending out messages) is finished successfully.
+In this mode the receive operation is wrapped in a transport native transaction. This mode guarantees that the message is not permanently deleted from the incoming queue until at least one processing attempt (including storing user data and sending out messages) is finished successfully. See the [errors](/nservicebus/errors) section for more details on retries.
 
-NOTE: In this mode messages on the wire can get duplicated at each endpoint so the handler logic has to be designed to be idempotent.
+Use the following code to request *Transport transaction* mode to be used:
 
 snippet:TransactionsDisableDistributedTransactions
 
+#### Multi queue transactions
+Some transports supports enlisting outgoing queue operations in the current receive transaction. This removes the risk of duplicate and ghost messages, see below, being emitted but you still need to make sure that your handlers are idempotent. Currently only the MSMQ and SQL Server transports support multi queue transactions.
 
-### Ambient transaction
+#### Consistency guarantees
 
-In this mode, if supported by the transport, the receive transaction is wrapped in `TransactionScope`. All the operations inside this scope, both sending messages and manipulating data, are guaranteed to be executed (eventually) as a whole or rolled back as a whole. There is no atomicity guarantee as changes to the database might be visible *before* the messages on the queue or vice-versa. Specific configurations might provide stronger guarantees, but not weaker.
+In this mode some (or all) handlers might get invoked multiple times and partial results might be visible:
 
-That **does not** mean that a distributed transaction is started right away. The transaction is only escalated to the distributed one (following two-phase commit protocol) when it is required. For example is not required when using SQL Server transport with NHibernate persistence and both in the same database. Since in this case ADO.NET driver guarantees that everything happens inside a single database transaction, ACID guarantees are held for the whole processing.
+ * partial updates - where one handler succeeded updating its data but the other didn't
+ * partial sends - where some of the messages has been sent but others not
 
-Ambient transaction mode is enabled by default. It can be enabled explicitly via
+To handle this you must make sure that you write code that is consistent from a business perspective even though its executed more than once. In other words you must ensure that your handlers are idempotent.
```

Comment:
Should I add a definition to http://docs.particular.net/nservicebus/concept-overview ?